domain/relation/service/relation.go

```diff
@@ -478,11 +538,272 @@ func (s *WatchableService) WatchApplicationSettings(
 
 // WatchLifeSuspendedStatus returns a watcher that notifies of changes to the life
 // or suspended status of the relation.
+//
+// The following error types can be expected to be returned:
+//   - [relationerrors.ApplicationIDNotValid] is returned if the application
+//     is not valid.
 func (s *WatchableService) WatchLifeSuspendedStatus(
+	applicationID application.ID,
+) (watcher.StringsWatcher, error) {
+	if err := applicationID.Validate(); err != nil {
+		return nil, errors.Errorf(
+			"%w: %w", relationerrors.ApplicationIDNotValid, err)
+	}
+	currentRelations := make(map[corerelation.UUID]relation.RelationLifeSuspendedData)
+	// returns a set of relation keys if the life or suspended status has changed
+	// for any relation this application is part of.
+
+	lifeTableName, suspendedTableName, query := s.st.InitialWatchLifeSuspendedStatus(applicationID)
+	return s.watcherFactory.NewNamespaceMapperWatcher(
+		func(ctx context.Context, txn database.TxnRunner) ([]string, error) {
+			relationUUIDStrings, err := query(ctx, txn)
+			if err != nil {
+				return nil, errors.Capture(err)
+			}
+
+			initialResults := make([]string, len(relationUUIDStrings))
```

Comment:
You can't do this, because you're using a continue, so you'd end up with a hole in the slice.